feat: Grunt mode post (#18)
R2Northstar/NorthstarBlog: _posts/2024-09-29-grunt-gamemode.md

---
title: "Grunt mode"
header:
  image: /assets/images/page-header-image.jpg
  og_image: /assets/images/page-header-image.jpg
tags:
  - Gamemode
author: EnderBoy9217
last_modified_at: 2024-09-29T13:18:00-01:00
---

Hello Northstar! I am EnderBoy9217 and here I will share with you the process of creating one of Northstar's most recognizable mods, Grunt Mode 2.

## How it started

> My friends and I have had this idea of this style of gameplay for a while. While we enjoy the fast pace of Titanfall, we also like the slower pace and more tactical gameplay of games like Battlefield or Squad at times, where instead of being a powerhouse, or a literal Titan, you were just another expendable soldier on the battlefield. Gamemodes like Titanfall 2’s attrition mode captured the chaos of the large-scale battles seen in these other games, however, it failed to fully immerse you into these as simply another soldier.

Reviewer: Weird apostrophe and newline for easier diffing
```suggestion
My friends and I have had this idea of this style of gameplay for a while. While we enjoy the fast pace of Titanfall, we also like the slower pace and more tactical gameplay of games like Battlefield or Squad at times, where instead of being a powerhouse, or a literal Titan, you were just another expendable soldier on the battlefield.
Gamemodes like Titanfall 2's attrition mode captured the chaos of the large-scale battles seen in these other games, however, it failed to fully immerse you into these as simply another soldier.
```

> ### Grunt Mode [1]

Reviewer: What's the purpose of the `[1]`? ^^"

Reviewer: Oh is it cause of Gruntmode v1?

Author: That's how I understood it :)
@EnderBoy9217 are we right?

> Many people have asked where the “2” in Grunt Mode 2 came from. The reason is that the initial creation of the Grunt Mode was not my own, but instead created by another modder on the platform, VoyageDB.

Reviewer: ```suggestion
Many people have asked where the "2" in Grunt Mode 2 came from.
The reason is that the initial creation of the Grunt Mode was not my own, but instead created by another modder on the platform, VoyageDB.
```

> VoyageDB’s new [Grunt Mode](<https://thunderstore.io/c/northstar/p/VoyageDB_Modding_Home/Grunt_Mode/>) gained massive popularity soon after its initial release. It featured an attrition match with more enemy types, including AI Titans, prowlers, and gunships, spawning in droppods, and best of all, everyone had the same grunt models and skins. It took many measures to slow down the gameplay, not only through the inclusion of new AI, but also through removing boosts, player titans, double jump, limiting wallrun distance, and later, removing weapon attachments. This mod also introduced the start of the classes mechanic, where you had a small chance to spawn with either two ticks that turned into your own personal plasma drones, or a shield captain’s mobile shield. This mode also featured care packages, which would occasionally land and give out weapons pre-fitted with attachments.

Reviewer: ```suggestion
VoyageDB's new [Grunt Mode](<https://thunderstore.io/c/northstar/p/VoyageDB_Modding_Home/Grunt_Mode/>) gained massive popularity soon after its initial release.
It featured an attrition match with more enemy types, including AI Titans, prowlers, and gunships, spawning in droppods, and best of all, everyone had the same grunt models and skins.
It took many measures to slow down the gameplay, not only through the inclusion of new AI, but also through removing boosts, player titans, double jump, limiting wallrun distance, and later, removing weapon attachments.
This mod also introduced the start of the classes mechanic, where you had a small chance to spawn with either two ticks that turned into your own personal plasma drones, or a shield captain’s mobile shield.
This mode also featured care packages, which would occasionally land and give out weapons pre-fitted with attachments.
```

## Adding new features
After establishing the mod as its own experience, I included many features that myself and other players believed would help ground the experience. One of the first being the ability for AI to score points for their team. While some of VoyageDB’s code comments alluded that they had attempted to do this in their own mod, this seemingly small change completely changed how the game worked. The game was no longer a battle of players but instead a battle of teams, with each individual AI unit playing their own part. This also led to many people downloading the mod simply to play a solo experience akin to games like Ravenfield, as now it was possible to lose a match, even if you were the only player, based on your performance.

> While I added many features, I also later removed some of VoyageDB’s work which I found didn’t fit in well with my new direction toward grounded gameplay. I removed the care package system, removing powerful weapons such as the smart pistol out of rotation, limiting grenades, removing the ability to rodeo, and as previously mentioned, wallrunning was removed entirely. While the intention of the limited wallrun was to allow for the climbing of walls, it felt out-of-place, and the client’s underlying visual and audio language made the intended feature feel more like a bug. However, while most maps were almost entirely accessible through doors and stairways, many locations like rooftops were now entirely out-of-play by players, slightly reducing the amount of variety in the game. This shortcoming was later solved with the development of the most recognizable feature of grunt mode, its new economy and class system. New gamemodes and grunt versions of existing gamemodes like hardpoint were also added to Grunt Mode 2 later in its development.

Reviewer: ```suggestion
While I added many features, I also later removed some of VoyageDB's work which I found didn’t fit in well with my new direction toward grounded gameplay.
I removed the care package system, removing powerful weapons such as the smart pistol out of rotation, limiting grenades, removing the ability to rodeo, and as previously mentioned, wallrunning was removed entirely.
While the intention of the limited wallrun was to allow for the climbing of walls, it felt out-of-place, and the client’s underlying visual and audio language made the intended feature feel more like a bug.
However, while most maps were almost entirely accessible through doors and stairways, many locations like rooftops were now entirely out-of-play by players, slightly reducing the amount of variety in the game.
This shortcoming was later solved with the development of the most recognizable feature of grunt mode, its new economy and class system.
New gamemodes and grunt versions of existing gamemodes like hardpoint were also added to Grunt Mode 2 later in its development.
```
